### PR TITLE
Make it possible to remove single history entry

### PR DESCRIPTION
**Pull request by rainemak**

### CODE REVIEW

**Comment by rainemak** [comment]
See also https://github.com/sailfishos/sailfish-browser/issues/253

@chriadam @siteshwar review? I won't merge this yet as this introduces new translation string but will do it right after translation round is over.

**Comment by rainemak** [comment]
@pvuorela review?

File: src/history/declarativehistorymodel.cpp [hunk 1]
```diff
@@ -39,6 +39,20 @@ void DeclarativeHistoryModel::clear()
     emit countChanged();
 }
 
+void DeclarativeHistoryModel::remove(int index)
+{
+    if (index < 0 || index > m_links.count()) {
```
Comment (pvuorela) [inline src/history/declarativehistorymodel.cpp]: Should be >= ?

File: tests/auto/tst_declarativehistorymodel/tst_declarativehistorymodel.cpp [hunk 2]
```diff
@@ -51,26 +52,43 @@ private slots:
...
+            .arg(QLatin1String(DB_NAME));
+    QFile dbFile(dbFileName);
+    dbFile.remove();
+}
+
+void tst_declarativehistorymodel::init()
+{
     tabModel = new PersistentTabModel(DBManager::instance()->getMaxTabId() + 1);
-    historyModel = TestObject::qmlObject<DeclarativeHistoryModel>("historyModel");
+    historyModel = new DeclarativeHistoryModel();
```
Comment (pvuorela) [inline tests/auto/tst_declarativehistorymodel/tst_declarativehistorymodel.cpp]: Extra ()

**Comment by pvuorela** [comment]
Overall LGTM

File: src/history/declarativehistorymodel.cpp (hunk 1 above)
Comment (rainemak) [inline src/history/declarativehistorymodel.cpp]: Yes it should, blindly copy-pasted a piece of old code. There seems to be a potential bug in data method as well. :)

**Comment by rainemak** [comment]
Fixed review findings and added one potential crash fix on top.

**Pull request merged**